src/TimeTravel.elm

```diff
@@ -3,3 +3,104 @@ module TimeTravel exposing (addTimeTravel)
 import Playground exposing (..)
 import Set
 
+controlBarHeight = 64
+maxVisibleHistory = 2000
+
+addTimeTravel rawGame = 
+  { initialState = initialStateWithTimeTravel rawGame
+  , updateState = updateWithTimeTravel rawGame
+  , view = viewWithTimeTravel rawGame
+  }
+
+initialStateWithTimeTravel rawGame = 
+  { rawModel = rawGame.initialState
+  , paused = False
+  , history = []
+  , historyPlaybackPosition = 0
+  }
+
+updateWithTimeTravel rawGame computer currentModel =
+  let
+    updatedModel = 
+      { currentModel
+      | rawModel = rawGame.updateState computer currentModel.rawModel
+      , history = currentModel.history ++ [computer]
+      , historyPlaybackPosition = currentModel.historyPlaybackPosition + 1
+      }
+    newPlaybackPosition = 
+      min
+        (mousePosToHistoryIndex computer)
+        (List.length currentModel.history)
+    replayHistory pastInputs = 
+      List.foldl 
+        (\input model -> rawGame.updateState input model)
+        rawGame.initialState
+        pastInputs
+  in
+    if currentModel.paused && computer.mouse.down then
+      { currentModel
+      | historyPlaybackPosition = newPlaybackPosition
+      , rawModel = replayHistory (List.take newPlaybackPosition currentModel.history)
+      }
+    else if keyPressed "R" computer then
+      { updatedModel
+      | paused = False
+      , history = List.take currentModel.historyPlaybackPosition currentModel.history
+      }
+    else if keyPressed "T" computer then
+      { currentModel | paused = True }
+    else if keyPressed "C" computer then
+      { currentModel
+      | rawModel = rawGame.initialState
```

Comment:
You did `initialStateWithTimeTravel rawGame`, instead of reassigning the whole model, which makes way more sense. I forgot that function existed. Also, you require that it be paused for pressing C to work, which I didn't, but I think I like it better that mine allows you to reset it anytime.